packages/aws-cdk/lib/context-providers/load-balancers.ts

```diff
@@ -0,0 +1,309 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import * as AWS from 'aws-sdk';
+import { Mode, SdkProvider } from '../api';
+import { ContextProviderPlugin } from './provider';
+
+/**
+ * Provides load balancer context information.
+ */
+export class LoadBalancerContextProviderPlugin implements ContextProviderPlugin {
+  constructor(private readonly aws: SdkProvider) {
+  }
+
+  async getValue(query: cxschema.LoadBalancerContextQuery): Promise<cxapi.LoadBalancerContextResponse> {
+    const elbv2 = (await this.aws.forEnvironment(cxapi.EnvironmentUtils.make(query.account, query.region), Mode.ForReading)).elbv2();
+
+    if (!query.loadBalancerArn && !query.loadBalancerTags) {
+      throw new Error('The load balancer lookup query must specify either `loadBalancerArn` or `loadBalancerTags`');
+    }
+
+    const loadBalancers = await findLoadBalancers(elbv2, query);
+
+    if (loadBalancers.length === 0) {
+      throw new Error(`No load balancers found matching ${JSON.stringify(query)}`);
+    }
+
+    if (loadBalancers.length > 1) {
+      throw new Error(`Multiple load balancers found matching ${JSON.stringify(query)} - please provide more specific criteria`);
+    }
+
+    const loadBalancer = loadBalancers[0];
+
+    const ipAddressType = loadBalancer.IpAddressType === 'ipv4'
+      ? cxapi.LoadBalancerIpAddressType.IPV4
+      : cxapi.LoadBalancerIpAddressType.DUAL_STACK;
+
+    return {
+      loadBalancerArn: loadBalancer.LoadBalancerArn!,
+      loadBalancerCanonicalHostedZoneId: loadBalancer.CanonicalHostedZoneId!,
+      loadBalancerDnsName: loadBalancer.DNSName!,
+      vpcId: loadBalancer.VpcId!,
+      securityGroupIds: loadBalancer.SecurityGroups ?? [],
+      ipAddressType: ipAddressType,
+    };
+  }
+}
+
+// Decreases line length
+type LoadBalancerListenerQuery = cxschema.LoadBalancerListenerContextQuery;
+type LoadBalancerListenerResponse = cxapi.LoadBalancerListenerContextResponse;
+
+/**
+ * Provides load balancer listener context information
+ */
+export class LoadBalancerListenerContextProviderPlugin implements ContextProviderPlugin {
+  constructor(private readonly aws: SdkProvider) {
+  }
+
+  async getValue(query: LoadBalancerListenerQuery): Promise<LoadBalancerListenerResponse> {
+    const elbv2 = (await this.aws.forEnvironment(cxapi.EnvironmentUtils.make(query.account, query.region), Mode.ForReading)).elbv2();
+
+    if (!query.listenerArn && !query.loadBalancerArn && !query.loadBalancerTags) {
+      throw new Error('The load balancer listener query must specify at least one of: `listenerArn`, `loadBalancerArn` or `loadBalancerTags`');
+    }
+
+    if (query.listenerArn) {
+      // When we know a listener arn, we can query for that listener directly.
+      return this.getListenerByArn(elbv2, query);
+    } else {
+      // When we don't know a listener arn, we need to find it by traversing
+      // all associated load balancers.
+      return this.getListenerByFilteringLoadBalancers(elbv2, query);
+    }
```

Comment:
`~` (Style) I'd probably just inline this:

```suggestion
    return query.listenerArn ? this.getListenerByArn(elbv2, query) : this.getListenerByFilteringLoadBalancers(elbv2, query);
```